debezium-core/src/main/java/io/debezium/transforms/ByLogicalTableRouter.java

```diff
@@ -127,58 +164,76 @@ public void configure(Map<String, ?> props) {
         if (keyFieldRegexString != null) {
             keyFieldRegexString = keyFieldRegexString.trim();
         }
-        if (keyFieldRegexString != null && keyFieldRegexString != "") {
+        if (keyFieldRegexString != null && !Objects.equals(keyFieldRegexString, "")) {
             keyFieldRegex = Pattern.compile(config.getString(KEY_FIELD_REGEX));
             keyFieldReplacement = config.getString(KEY_FIELD_REPLACEMENT);
         }
         keyFieldName = config.getString(KEY_FIELD_NAME);
-
-        keySchemaUpdateCache = new SynchronizedCache<>(new LRUCache<Schema, Schema>(16));
-        envelopeSchemaUpdateCache = new SynchronizedCache<>(new LRUCache<Schema, Schema>(16));
     }
 
     @Override
     public R apply(R record) {
         final String oldTopic = record.topic();
         final String newTopic = determineNewTopic(oldTopic);
+
         if (newTopic == null) {
             return record;
         }
 
-        logger.debug("Applying topic name transformation from " + oldTopic + " to " + newTopic + ".");
+        logger.debug("Applying topic name transformation from {} to {}", oldTopic, newTopic);
+
         final Struct oldKey = requireStruct(record.key(), "Updating schema");
         final Schema newKeySchema = updateKeySchema(oldKey.schema(), newTopic);
         final Struct newKey = updateKey(newKeySchema, oldKey, oldTopic);
 
         if (record.value() == null) {
             // Value will be null in the case of a delete event tombstone
-            return record.newRecord(newTopic, record.kafkaPartition(), newKeySchema, newKey, record.valueSchema(),
-                    record.value(), record.timestamp());
+            return record.newRecord(
+                    newTopic,
+                    record.kafkaPartition(),
+                    newKeySchema,
+                    newKey,
+                    record.valueSchema(),
+                    record.value(),
+                    record.timestamp()
+            );
         }
 
         final Struct oldEnvelope = requireStruct(record.value(), "Updating schema");
         final Schema newEnvelopeSchema = updateEnvelopeSchema(oldEnvelope.schema(), newTopic);
         final Struct newEnvelope = updateEnvelope(newEnvelopeSchema, oldEnvelope);
-        return record.newRecord(newTopic, record.kafkaPartition(), newKeySchema, newKey, newEnvelopeSchema,
-                newEnvelope, record.timestamp());
+
+        return record.newRecord(
+                newTopic,
+                record.kafkaPartition(),
+                newKeySchema,
+                newKey,
+                newEnvelopeSchema,
+                newEnvelope,
+                record.timestamp()
+        );
     }
 
     @Override
     public void close() {
-        keySchemaUpdateCache = null;
```

Comment:
I moved the cache instance creation to the respective member and made it final. There is one distinct SMT instance per connector (a second, third, ... per connector is theoretically possible, but rather useless). This leads to a potentially warm cache after e.g. a connector restart due to a reconfiguration. Otherwise every config change would invalidate the cache, independent of the SMT being affected by the config change or not.